builds/jenkins-build-pack-api/Jenkinsfile

```diff
@@ -830,41 +861,45 @@ def updateOperationId(swaggerFileContent) {
 def writeServerlessFile(config, environment_logical_id, accountDetails){
 
   def iamRoleArnValue;
-  if(config['domain'] == 'jazz'){
-    iamRoleArnValue = config['iamRoleARN']
-  } else {
-    iamRoleArnValue = accountDetails.IAM.USERSERVICES_ROLEID;
-  }
+  if(config['domain'] == 'jazz') iamRoleArnValue = config['iamRoleARN']
+  else iamRoleArnValue = accountDetails.IAM.USERSERVICES_ROLEID;  
 
   sh "sed -i -- 's/\${file(deployment-env.yml):service}/${configLoader.INSTANCE_PREFIX}-${config['domain']}-${config['service']}/g' serverless.yml"
   sh "sed -i -- 's/{inst_stack_prefix}/${configLoader.INSTANCE_PREFIX}/g' serverless.yml"
   sh "sed -i -- 's/{env-stage}/${environment_logical_id}/g' serverless.yml"
   sh "sed -i -- 's/\${file(deployment-env.yml):region}/${config['region']}/g' serverless.yml"
   sh "sed -i -- 's/\${file(deployment-env.yml):domain, self:provider.domain}/${config['domain']}/g' serverless.yml"
   sh "sed -i -- 's/\${file(deployment-env.yml):owner, self:provider.owner}/${config['created_by']}/g' serverless.yml"
-  sh "sed -i -e 's|\${file(deployment-env.yml):iamRoleARN}|${iamRoleArnValue}|g' serverless.yml"
   sh "sed -i -- 's/\${file(deployment-env.yml):providerRuntime}/${config['providerRuntime']}/g' serverless.yml"
   sh "sed -i -- 's/\${file(deployment-env.yml):providerMemorySize}/${config['providerMemorySize']}/g' serverless.yml"
   sh "sed -i -- 's/\${file(deployment-env.yml):providerTimeout}/${config['providerTimeout']}/g' serverless.yml"
-  sh "sed -i -- '/#Start:resources/,/#End:resources/d' ./serverless.yml"
   sh "sed -i -- '/#Start:isScheduleEnabled/,/#End:isScheduleEnabled/d' ./serverless.yml"
 
   if ((config['service'] in configLoader.JAZZ.VPC_SERVICES.APIS) && config['domain'] == 'jazz' && configLoader.JENKINS.DOCKERIZED == true) {
   	addVpcDetails(config)
   }
+
   if (config['artifact']) {
     sh "sed -i -- 's/\${file(deployment-env.yml):artifact}/${config['artifact']}/g' serverless.yml"
   }
+
   if (config['mainClass']) {
     sh "sed -i -- 's/\${file(deployment-env.yml):mainClass}/${config['mainClass']}/g' serverless.yml"
   }
-}
 
-/**
-	Set the Lambda execution role. If user provided a custom role override the default.
-*/
-def setLambdaExecutionRole(role) {
-  sh "sed -i -- 's#DEFAULT_LAMBDA_EXE_ROLE#${role}#g' serverless.yml"
+  if (config['domain'] != 'jazz' && configLoader.TVAULT && configLoader.TVAULT.IS_ENABLED ) {
```

Comment:
I think this can be simplified further. As a best practice, we can create dedicated roles per service/environment irrespective of whether vault feature is enabled or not. 

If vault is enabled, add role to the safe
If vault is disable, do nothing.

cc @raghits

Reply:
We can make this change after merging all the open PRs for secret management.